src/core/metrics.cc

```diff
@@ -272,6 +272,36 @@ void impl::remove_registration(const metric_id& id) {
     }
 }
 
+void impl::replicate_metric_if_required(shared_ptr<registered_metric> metric) const {
+    auto full_name = metric->get_id().full_name();
+
+    auto matching_spec = std::find_if(
+        _metrics_to_replicate.begin(),
+        _metrics_to_replicate.end(),
+        [&full_name](const auto& spec) {
+          return full_name == spec.full_name;
+        });
+
+    if (matching_spec != _metrics_to_replicate.end()) {
+        const auto& metric_family = _value_map.at(full_name);
+        replicate_metric(*matching_spec, metric, metric_family);
+    }
+}
+
+void impl::remove_metric_replica_if_required(const metric_id& id) const {
+    auto matching_spec = std::find_if(
+        _metrics_to_replicate.begin(),
+        _metrics_to_replicate.end(),
+        [full_name = id.full_name()](const auto& spec) {
+          return full_name == spec.full_name;
+        });
+
+    if (matching_spec != _metrics_to_replicate.end()) {
+        auto destination_impl = get_local_impl(matching_spec->destination_handle);
```

Comment:
Hold by ref?

Reply:
I don't mind. It's a `shared_ptr` so the cost of copying it shouldn't be too bad.